File: src/main/java/com/example/form/TaskForm.java
package com.example.form;

import com.example.domain.Task;
import lombok.AllArgsConstructor;
import lombok.Data;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;
import javax.validation.constraints.Size;

@Data
@AllArgsConstructor
public class TaskForm {
    @NotNull
    @Size(max = 32)
    private String title;

    @Size(max = 255)
    private String description;

    @NotNull
    @Pattern(regexp = "OPEN|CLOSE|CANCEL")
    private String status;

    public Task.Status getStatusType() {
        return Task.Status.Companion.getStatus(this.status);
    }
}

File: src/main/java/com/example/form/TaskForm.kt
package com.example.form

import com.example.domain.Task
import javax.validation.constraints.NotNull
import javax.validation.constraints.Pattern
import javax.validation.constraints.Size

data class TaskForm(
        @field:NotNull @field:Size(max = 32) var title: String? = null,
        @field:Size(max = 255) var description: String? = null,
        @field:NotNull @field:Pattern(regexp = "OPEN|CLOSE|CANCEL") var status: String? = null
) {
    fun getStatusType(): Task.Status? {
        return Task.Status.getStatus(status)
    }
}
